core/src/connection/pool.rs

```diff
@@ -551,6 +592,20 @@ where
         TConnInfo: ConnectionInfo<PeerId = TPeerId> + Clone,
         TPeerId: Clone
     {
+        // Drain events resulting from forced disconnections.
+        while let Some(Disconnected {
+            id, connected, num_established
+        }) = self.disconnected.pop() {
```

Comment:
A comment about the importance of `.pop()` w.r.t. the correctness of `num_established` would be appreciated.